✅ Update Actions to latest versions

thisiskeithb opened the pull request:
### Description

- Update Actions to latest versions because *most* are using Node.js 20 now due to Node.js 16 being deprecated. See [GitHub Blog - GitHub Actions: Transitioning from Node 16 to Node 20](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/) for more information.

   _Note: `actions-ecosystem/action-remove-labels` & `superbrothers/close-pull-request` still need to update their Actions for Node.js 20 compatibility._
- `lock-threads` Action replaced `lock-threads-app` "Application", so `lock.yml` is no longer needed.

thisiskeithb committed:
✅ Update Actions to latest versions

classicrocker883 commented:
it appears the parameters have changed as well for [dessant/lock-thread](https://github.com/dessant/lock-threads) check out the readme for changes.  
for instance:
```diff
-   issue-lock-inactive-days: '60'
-   issue-exclude-created-before: ''
+   issue-inactive-days: '60'
+   exclude-issue-created-before: ''
```

thisiskeithb committed:
lock-threads Action replaced lock-threads-app "Application"

This is no longer needed. lock-threads-app is also archived & no longer maintained.
